### DISCUSSION TITLE
What to do with the unconcealed/unblinded datavector?

### DISCUSSION THREAD

**Discussion by arthurmloureiro** [LSSTDESC/Smokescreen#52] [Q&A]
We should discuss what to do with the unblinded data-vectors.

When appliying smokescreen to something like TXPipe, we can manipulate the original datavector and save to disk only the concealed datavector. But what should we do with the original one?

- Should we provide tools to encrypt/decrypt the original datavector? 
> In this case, we provide these tools in smokescreen so everything is self-contained

- Should we implement safeguards in SACC to avoid someone accidentally reading the unconcealed data-vector?
> In this case, we need to coordinate with SACC to create this metadata informing SACC that it is handling the unblinded datavector but then how to ensure/stop the user from accidentally opening it? 

- Maybe something else?

Tagging @jablazek @jessmuir and @joezuntz for the discussion.
Feel free to tag more people :)

**Comment by joezuntz** [comment]
In the current TXPipe version we have been deleting (actually overwriting with a blank file) the unblinded version, just to be absolutely sure.  We could also encrypt it instead, sure, that would save re-running when we want to unblind.

**Reply by arthurmloureiro** [reply]
Yeah, that's what I had in mind... if we encrypt it, we avoid accidental unblinding. 

I don't think we must worry that much about what to do with the encryption key, right?
I can add a giant path to find it `./unblind_key/are_you_sure/have_you_checked_all_systematics/what_about_b_modes/ok_then/have_your_key/encryption.key` 🤣  

and so smokecreen can have a decrypt function where you provide a path to the encrypted sacc and the key to decrypt it. 

If we are afraid people may be malicious and unblind if they have the key... well... then we can discuss if we email it to René and no one else hahaha

**Reply by jablazek** [reply]
Since we are talking about something that can be regenerated, we don't need to worry too much about the key being lost. I guess if we are going to throw it away, we want to make sure the TXPipe config file is stored in an easy to find place.

**Reply by arthurmloureiro** [reply]
@jablazek personally I'd have no problem with simply saving the key to the same directory as the data-vector is saved from TXPipe and we can trust everyone is a responsible ethical scientist and won't unblind before time. This approach prevents accidental unblinding, since you'd have to manually decrypt the file, which I think is the main goal.

But I imagine others in the collaboration could disagree with leaving the key around ...
